ui/appui-react/src/appui-react/stagepanels/StagePanelDef.tsx

```diff
@@ -116,7 +125,32 @@ export class StagePanelDef extends WidgetHost {
   public get resizable(): boolean { return this._resizable; }
 
   /** Indicates whether the panel is pinned. */
-  public get pinned(): boolean { return this._pinned; }
+  public get pinned(): boolean {
+    if (UiFramework.frontstages.activeFrontstageDef) {
+      const state = UiFramework.frontstages.activeFrontstageDef.getPanelCurrentState(this);
+      return state[2];
+    }
+    // istanbul ignore next
+    return false;
+  }
+
+  public set pinned(pinned: boolean) {
+    if (this._pinned === pinned)
+      return;
```

Comment:
Why is the getter relying on external state if the setter do not ?

If the internal state is not reliable for the getter, maybe the setter should not have an internal state at all ? (and rely on the getter)

Reply:
A fair point. As I see it, the sole purpose of having this internal state is to confirm if the state's value has actually become out of sync with its external source (since setters are always called on all `StagePanelsDef`s whenever one of them changes) and needs to emit some onChange event. Additionally, I just reused code of `size` and `panelState`. So, if I were to rely only on external getter, I wouldn't know whether or not the onChange event should be emited.
A possible solution could be to shift responsibility of keeping tract of the previous `StagePanelsDef`s states onto code that triggers these setters (namely `FrontstageDef` as it already does so), but the code would get way more cumbersome and I wouldn't be sure if it would be the better pattern here since the internal state of all these props were already here even before my changes :).
Sooo... Any suggestions?

Reply:
You could have just said: Other setters are working like this in this class :D

Alright, that seems off but for this PR let stick to the standard of this file.